jobspec/parse.go

```diff
@@ -195,6 +195,7 @@ func parseConstraints(result *[]*api.Constraint, list *ast.ObjectList) error {
 			}
 
 			m["Operand"] = api.ConstraintDistinctHosts
+			m["RTarget"] = strconv.FormatBool(enabled)
```

Comment:
Is there a specific test requirement for this?  It is defaulted in scheduler/feasible.go (576-578) to true. Not a big deal and hard to see around it with phone review.

Reply:
It actually seems to break `TestParse/distinctHosts-constraint.hcl `

Reply:
Yes, since the test explicitly sets `true` the hcl2 parser in jobspec2 explicitly sets `true`, so this change to the hcl1 parser (we really gotta drop this thing) makes it match the hcl2 behavior.